slapp/transforms/image_utils.py

```diff
@@ -1,76 +1,56 @@
-from typing import Tuple, Union
-from pathlib import Path
-from math import ceil
-
+from typing import Tuple
 import numpy as np
-from PIL import Image, ImageDraw, ImageFont
+import cv2
 
 
-def add_scale(image: Union[Path, np.ndarray],
+def add_scale(array: np.ndarray,
               scale_position: Tuple[int, int],
-              font_file: Path,
-              resolution: float = 400 / 512,
-              scale_size: int = 10,
-              fill: Tuple[int, int, int] = (0, 0, 0),
-              width: int = 5,
-              font_size: int = 8) -> Image:
+              um_per_pixel: float = 400 / 512,
+              scale_size_um: float = 10,
+              color: int = 1,
+              thickness_um: float = 1.0,
+              fontScale: float = 0.2) -> np.ndarray:
     """
-    Adds a scale bar onto a image passed in the form of a image path or
-    a numpy array. Uses PIL library ImageDraw and ImageFont to draw scale bar
-    onto images.
+    Adds a scale bar onto an array using opencv.
+
     Parameters
     ----------
-    image: [Path, np.ndarray]
-        path to the image to add scale bar
+    array: np.ndarray
+        function only accepts np.uint8 2D arrays
     scale_position: Tuple
-        the position at which to set the bottom left corner of the scale. In
-        x, y format, this is the format PIL expects.
-    font_file: Path
-        the font file to be used as denoted by a path object. On windows can
-        just be specified as 'arial.ttf' on linux must be a full path.
-    resolution: float = 10/13
-        resolution of the video in microns per pixel, defaulted to
-        scientifica general which is 10/13~.78 micros/pixel
-    scale_size: int=10
-        how many micro meters the scale bar should display, defaulted to 10
-    fill: Tuple=(0, 0, 0)
-        This parameter is passed to line draw in PIL libary and name reflects
-        name in that API. The fill color in rgb tuple to draw the text and
-        scale, defaults to black
-    width: Int
-        This parameter is passed to line draw in PIL libary and name reflects
-        name in that API. The width of the lines of the scale in pixels
-    font_size: Int
-        the font size of the description text
+        image coordinates for the corner of the scale bars
+    um_per_pixel: float = 400/512
+        resolution of the video in microns per pixel
+    scale_size_um: float
+        length in microns of scale bars
+    color: int
+        default 1. passed as `color` to cv2.line and cv2.putText
```

Comment:
Would appreciate more explanation here, so I don't have to look up what color is in cv2. Is it a 0-1 scale, where 1 is black?

Reply:
I'm not sure where this came about in our development, but, the front-end wants a black line on a transparent background. Black is `0` and white is `255`. We are setting `transparency=0` so we can't really draw in black. So I set it to 1 (or it was set to 1, can't remember).
The right way to fix this is for `generate_ROI_outline` to return a black (`0`) outline on a white (`255`) background and then tell `imageio` `transparency=255`. I'm happy to make that change.

Reply:
What you described would make more sense. 

The reason the front-end uses black is because it uses color rotation to produce different colors for colorblind mode vs. default colors, rotated from black (which is a fairly straightforward computation).

Reply:
implemented